### PR TITLE
Stats: Swap date display date range and title

### PR DESCRIPTION
**Pull request by kangzj**
Addresses feedback from p1HpG7-v6l-p2

## Proposed Changes

* Swap the logic to get display date for title and date range picker
* Show container even when no content to avoid jumpy content when switching dates

## Why are these changes being made?

* Address feedback

## Testing Instructions

-  Open Calypso Live on the Traffic page for a p2 or internal site `/stats/day/:siteSlug`
- Choose a shortcut from date range picker
- Ensure the title show the shortcut label while the date range shows the exact dates
- Ensure the content isn't jumpy when switching dates/ranges

 
## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/date-range/use-shortcuts.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	const getShortcutForRange = () => {
+		if ( ! dateRange?.chartEnd || ! dateRange?.chartStart ) {
+			return null;
+		}
+		// Search the shortcut array for something matching the current date range.
+		// Returns shortcut or null;
+		const today = siteToday.format( 'YYYY-MM-DD' );
+		const yesterday = siteToday.clone().subtract( 1, 'days' ).format( 'YYYY-MM-DD' );
+		const shortcut = supportedShortcutList.find( ( element ) => {
```
Comment (kangzj) [inline client/components/date-range/use-shortcuts.ts:100]: The matching function needs future work, as it is not flexible enough to support shortcuts like `Last month` etc.

File: client/my-sites/stats/stats-date-picker/index.jsx [hunk 2]
```diff
@@ -171,8 +173,10 @@ class StatsDatePicker extends Component {
 
 	render() {
 		/* eslint-disable wpcalypso/jsx-classname-namespace*/
-		const { summary, translate, query, showQueryDate, isActivity, isShort } = this.props;
+		const { summary, translate, query, showQueryDate, isActivity, isShort, dateRange, reduxState } =
+			this.props;
 		const isSummarizeQuery = get( query, 'summarize' );
+		const { selectedShortcut } = getShortcuts( reduxState, dateRange, undefined, translate );
```
Comment (dognose24) [inline client/my-sites/stats/stats-date-picker/index.jsx]:
```suggestion
		const { selectedShortcut } = getShortcuts( reduxState, dateRange, undefined, translate, isNewDateFilteringEnabled );
```
We might need to pass `isNewDateFilteringEnabled` from the [ancestor](https://github.com/Automattic/wp-calypso/blob/26dbfeaedd1434a405b695b9980baf4e3e2f795e/client/my-sites/stats/site.jsx#L485-L493) for internal sites check for Today and Yesterday.

截圖 2024-12-02 下午2 45 29

**Review by dognose24 — APPROVED** [review]
LGTM! However, the `Today` and `Yesterday` shortcuts are not working.

截圖 2024-12-02 下午2 45 29

File: client/my-sites/stats/stats-date-picker/index.jsx (hunk 2 above)
Comment (kangzj) [reply client/my-sites/stats/stats-date-picker/index.jsx]: Good catch!

**Pull request merged**